build.gradle.kts

```diff
@@ -1,22 +1,27 @@
+import org.jetbrains.kotlin.k2.blogpost.DownloadGradleProfileTask
 import org.jetbrains.kotlin.k2.blogpost.PerformanceTask
 
 plugins {
     id("org.jetbrains.kotlin.jvm") version "1.9.23"
     `kotlin-dsl`
 }
 
+val downloadGradleProfileTask = DownloadGradleProfileTask.registerTask(project)
+
 val cleanGeneratedReports_2_0 = tasks.register<Delete>("cleanGeneratedReports2_0") {
     delete(layout.projectDirectory.dir("reports/2.0.0-RC1"))
 }
 val benchmark_2_0 = PerformanceTask.registerPerformanceTask(project, "benchmark_2_0", "2.0.0-RC1") {
     dependsOn(cleanGeneratedReports_2_0)
+    dependsOn(downloadGradleProfileTask)
```

Comment:
Nit: you could wire output of `DownloadGradleProfilerTask` to input of `PerformantTask.gradleProfilerDir` and then Gradle should automatically figure out task dependencies. Also, I propose to pass `TaskProvider<DownloadGradleTask>` as additional parameter to the `registerPerformanceTask()` method.